packages/rs-dpp/src/identity/state_transition/identity_credit_withdrawal_transition/apply_identity_credit_withdrawal_transition_factory.rs

```diff
@@ -34,37 +40,73 @@ where
         &self,
         state_transition: &IdentityCreditWithdrawalTransition,
     ) -> Result<()> {
-        let latest_withdrawal_index = self
+        let data_contract_id = Identifier::from_string(
+            &withdrawals_contract::system_ids().contract_id,
+            Encoding::Base58,
+        )?;
+        let data_contract_owner_id = Identifier::from_string(
+            &withdrawals_contract::system_ids().owner_id,
+            Encoding::Base58,
+        )?;
```

Comment:
You should also call the system_ids() only once and extract the contract_id and owner_id, before these functions.